kythe/java/com/google/devtools/kythe/extractors/java/JavaCompilationUnitExtractor.java

```diff
@@ -791,6 +796,17 @@ private AnalysisResults runJavaAnalysisToExtractCompilationDetails(
             public void finished(TaskEvent e) {
               if (e.getKind() == TaskEvent.Kind.PARSE) {
                 compilationUnits.add(e.getCompilationUnit());
+                try {
+                  String annotationPath =
+                      e.getCompilationUnit().getSourceFile().toUri().getPath() + ".pb.meta";
+                  for (JavaFileObject file :
+                      fileManager.getJavaFileForSources(Arrays.asList(annotationPath))) {
+                    ((UsageAsInputReportingJavaFileObject) file).markUsed();
+                    ((UsageAsInputReportingJavaFileObject) file).markOptional();
```

Comment:
This whole extra annotation is unnecessary if you just check whether or not the file exists before marking it as used.

Reply:
What's the recommended way to test this? FileObject#getLastModified?

Reply:
`Files.exists(annotationPath)` I would think.

Reply:
Or `Files.exists(Paths.get(annotationPath))`.  When we can actually use JDK9+ in source it gets a bit simpler as we don't have to round-trip through String/URI any more.